### DISCUSSION TITLE
Can I use PCM to collect PID and Virtual Memory that triggers TLB miss?

### DISCUSSION THREAD

**Discussion by twicy** [opcm/pcm#487] [Q&A] [answered]
Hi, I am new to the community, and hopefully I don't by any chance break any rules.

My aim: track down the virtual address of pages that trigger tlb miss and l3 miss at process-level

It seems to me that PCM is system-level? Is it possible that PCM outputs sampling records with PID and virtual memory
address target for the memory access events?

Many thanks in advance!

**Comment by rdementi** [comment, marked as answer]
PCM does not support sampling. I believe on Linux it is possible to achieve your goal with "perf record" & "perf script".

**Reply by twicy** [reply]
Many thanks to your generous advice! The questions to follow may be irrelavent to intel pcm, but

Just digging a little bit deeper, using perf will give me a .data file that I can read with "perf report", including the virtual memory address I desire, but what I really want is to write those addresses to a buffer.
Will it be a good idea to look through perf's source code and make some modifications, so that I will output the result to the buffer that I wanted, but not the .data file perf orignially appointed?

**Comment by opcm** [comment]
instead of "perf report" use "perf script". It allows to dump all fields, including addresses (--fields options) . perf script --help
